### PR TITLE
E2E Tests: stabilize Percy snapshot by avoiding random file names

### LINKED ISSUE #6665: SVG Upload: unstable Percy snapshot due to random image name
## Task Description

Uploaded images get random file names. This leads to this Percy snapshot always changing due to the random file name.

We should probably just not use random image names. There's no reason they _have to_ be random.

**Example:**

https://percy.io/google/web-stories-wp-e2e/builds/9216440/changed/521740708

Screenshot 2021-03-08 at 16 46 36

**Aside:** why is the SVG inserted at such a small size?

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Less random names to media elements.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6665

### CODE REVIEW

File: packages/e2e-test-utils/src/uploadFile.js [hunk 1]
```diff
@@ -20,8 +20,11 @@
 import { join, extname, resolve } from 'path';
 import { tmpdir } from 'os';
 import { copyFileSync } from 'fs';
-import { v4 as uuid } from 'uuid';
 
+const hashCode = (s) =>
+  s.split('').reduce((a, b) => ((a << 5) - a + b.charCodeAt(0)) | 0, 0);
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/uploadFile.js]: This looks like it was copied from somewhere, seems a bit magic without an explanation of what this does.

Why can't we just use the file name as-is without doing any such magic?

Comment (spacedmonkey) [reply packages/e2e-test-utils/src/uploadFile.js]: I don't know, it doesn't seem to like, which is why I closed the other PR and opened this.

**Comment by spacedmonkey** [comment]
Opened a new PR, as the last one was impossible to debug why it was not working locally.

**Comment by spacedmonkey** [comment]
The reason a random name was used in the first place, was because the play svg was imported on startup. It means when you upload it again, gets a prefix. Giving it a random name, means, when you go to delete it, then it no conflict or issue.

File: packages/e2e-test-utils/src/uploadFile.js (hunk 1 above)
Comment (spacedmonkey) [reply packages/e2e-test-utils/src/uploadFile.js]: Copied from https://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript

**Comment by swissspidy** [comment]
> The reason a random name was used in the first place, was because the play svg was imported on startup. It means when you upload it again, gets a prefix. Giving it a random name, means, when you go to delete it, then it no conflict or issue.

Can we avoid this by simply avoid this by not uploading images that we've already imported?

So we import images A, B, C for use in tests.

For uploading, we only ever upload D, E, F and delete them right afterwards.

**Comment by spacedmonkey** [comment]
> Can we avoid this by simply avoid this by not uploading images that we've already imported?

So I did this here. But tests still failed. 
https://github.com/google/web-stories-wp/pull/6680/commits/6b7cd9a86cfba0332a2eee05e9313b7e08011d99

For some reason, it just doesn't like reusing a file name. Maybe because the file goes into trash on delete and the slug is still reserved. I don't see the problem with changing the name of the file here? The difference is using the has means it is a more predictable new name. 

I haven't spent a lot of time on this ticket as it is, only 2 points. I have not be able to replicate this locally, so debug is waiting on E2E runs to complete in CI. With issues with CI on release work to do, it is not a focus. 

So

1. EIther we up the points and spend more times on this ticket to work the reason it doesn't like reusing names. 
2. Go with current solution that works and loop around to debug this later.

File: packages/e2e-test-utils/src/uploadFile.js (hunk 1 above)
Comment (swissspidy) [reply packages/e2e-test-utils/src/uploadFile.js]: We can't just copy code from SO for copyright/license reasons.

**Comment by swissspidy** [comment]
> > Can we avoid this by simply avoid this by not uploading images that we've already imported?
> 
> So I did this here. But tests still failed.
> 6b7cd9a86cfba0332a2eee05e9313b7e08011d99

I don't see how 6b7cd9a86cfba0332a2eee05e9313b7e08011d99 would have made a difference 🤔 

Neither `arrow.svg` or `video-play.svg` is used for uploads.

At the moment we import the following files:

* `small-video.webm` (with poster `small-video-poster.jpg`)
* `example1.jpg`
* `example2.jpg`
* `example3.jpg`
* `video-play.svg`

In E2E tests we currently upload the following files:

* `example1.jpg` (2 times
* `small-video.webm` (3 times)
* `close.svg` (1 time)

What I was suggesting is instead of uploading `small-video.webm` and `example1.jpg` in tests we could upload other files, let's say `new-video.mp4`, and `example4.jpg`.

Alternatively, we could do something like `uploadMedia(fileName, desiredNewFileName)`

> For some reason, it just doesn't like reusing a file name. Maybe because the file goes into trash on delete and the slug is still reserved.

`MEDIA_TRASH` is disabled in tests, fwiw.

> I don't see the problem with changing the name of the file here? The difference is using the has means it is a more predictable new name.

It might be predictable but less readable. Besides, it uses copied code with unknown license.

> I haven't spent a lot of time on this ticket as it is, only 2 points. I have not be able to replicate this locally, so debug is waiting on E2E runs to complete in CI. With issues with CI on release work to do, it is not a focus.
> 
> So
> 
> 1. EIther we up the points and spend more times on this ticket to work the reason it doesn't like reusing names.
> 2. Go with current solution that works and loop around to debug this later.

Let's do 1), but can of course do this after more pressing things have been fixed.

Can also do 2) but need to swap out the function with something else

**Comment by spacedmonkey** [comment]
I wish, I had more time to look into this, but as is, I have already spent a long time looking into an issue that isn't that important to the project as a whole. 

I can't workout why it doesn't like using the existing name, it doesn't make a lot of sense to me. But again, the time to work it out compared to what we would gain doesn't make a lot of sense to me. 

If we go with option 2, then all we need to do, is have a function that takes an string and replace a hashed or encoded version of that string that is predictable. Creating a function like that should be simple enough. 

Again if anyone else can work this out, then fine but I don't want to send that much more time on this..

**Review by swissspidy — APPROVED** [review]

**Pull request merged**